### PR TITLE
Migrate CI from Travis CI to GitHub Actions

### PR DESCRIPTION
**Pull request by adrianomitre**
This pull request migrate the CI from Travis CI to GitHub Actions, as requested by José Valim in [this comment](https://github.com/elixir-plug/plug/pull/957#issuecomment-647738665).

In addition, it makes sure all recommended Elixir and Erlang/OTP combinations are run on CI.

Quoting José Valim:
> I would still advise folks to go with:
> * For each Elixir version, add a run with earliest supported Erlang/OTP
> * For the last Elixir version, also test the latest supported Erlang/OTP

Please refer to the original discussion on https://github.com/dashbitco/broadway/pull/188 for the full rationale.

### CODE REVIEW

**Comment by ericmj** [comment]
This looks good but we don't have to test all OTP versions so please use the latest compatible OTP version for every Elixir version we test.

**Comment by adrianomitre** [comment]
The proposed solution does not test all OTP versions, just the earliest supported Erlang/OTP version for each Elixir version, except for the last Elixir version, in which case it also tests the latest supported Erlang/OTP, as suggested by José Valim as a compromise between completeness and resource efficiency.

**Comment by ericmj** [comment]
> Are you proposing to test only the latest, instead of the earliest, OTP version for each Elixir version? This would only reduce the total number of tests by one. Would you mind explaining why do you prefer that?

Yes, that's what I am proposing. It is unlikely that there will be any OTP specific issues with Decimal since it does not use anything from OTP directly. If there would be an issue with a specific OTP version it would therefor probably be an Elixir issue and I think it's unlikely we will find any such issues with Elixir versions that are several years old.

In fact I would even go so far to only test the latest and oldest supported Elixir version: 1.10.4 and 1.2.6.

I don't think we should waste and abuse resources that companies such as GitHub and Travis provide to us for free. If users find any issues with other specific Elixir and OTP versions then we can add specific tests for that and add them to CI if that happens.

**Comment by wojtekmach** [comment]
Agreed about keeping the matrix small, running on earliest and the current latest Elixir seems like a pragmatic choice here.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          - elixir: 1.6.6
+            otp: 21.3.8.16
+          - elixir: 1.7.4
+            otp: 22.3.4.4
+          - elixir: 1.8.2
+            otp: 22.3.4.4
+          - elixir: 1.9.4
+            otp: 22.3.4.4
+          - elixir: 1.10.4
+            otp: 23.0.3
```
Comment (ericmj) [inline .github/workflows/ci.yml:34]:
```suggestion
          - elixir: 1.2.6
            otp: 18.3.4.11
          - elixir: 1.10.4
            otp: 23.0.3
```

**Comment by ericmj** [comment]
Thank you!

**Pull request merged**